File: 26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#
# @lc code=start
class Solution:
    def removeDuplicates(self, nums):
    # def removeDuplicates(self, nums: List[int]) -> int:
        index = len(nums)-1
        if not len(nums):
            return 0
        if not len(nums)-1:
            return 1
        while index > 0:
            if nums[index]==nums[index-1]:
                nums.remove(nums[index])
            index=index-1
        return len(nums)
# @lc code=end

if __name__ == "__main__":
    print(Solution().removeDuplicates([0,0,1,1,1,2,2,3,3,4],))

File: 26.remove-duplicates-from-sorted-array.go
/*
 * @lc app=leetcode id=26 lang=golang
 *
 * [26] Remove Duplicates from Sorted Array
 */
package main

// @lc code=start
func removeDuplicates(nums []int) int {
	if len(nums) <= 1 {
		return len(nums)
	}

	left := 1
	right := 1
	for ; right < len(nums); right++ {
		if nums[right] != nums[right-1] {
			nums[left] = nums[right]
			left++
		}
	}
	return left
}

// @lc code=end
